File: 0961-n-repeated-element-in-size-2n-array/0961-n-repeated-element-in-size-2n-array.cpp
class Solution {
public:
    int repeatedNTimes(vector<int>& nums) {
        int n2 = nums.size();
        vector<int> freq(10001, 0);

        for(int num : nums) {
            if(++freq[num] == n2/2) {
                return num;
            }
        }

        return -1;
    }
};

File: 0961-n-repeated-element-in-size-2n-array/0961-n-repeated-element-in-size-2n-array.go
func repeatedNTimes(nums []int) int {
    n2 := len(nums)
    freq := make([]int, 10001) 
    for _, num := range nums {
        freq[num]++ 
        if freq[num] >= n2/2 {
            return num
        }
    }

    return -1
}